src/main/java/org/sunbird/bpreports/service/BPReportConsumer.java

```diff
@@ -0,0 +1,457 @@
+
+        String courseId = (String) request.get(Constants.COURSE_ID);
+        String batchId = (String) request.get(Constants.BATCH_ID);
+        String orgId = (String) request.get(Constants.ORG_ID);
+        String profileSurveyId = (String) request.get(Constants.PROFILE_SURVEY_ID);
+        Workbook workbook = new XSSFWorkbook();
+        Map<String, Object> headerKeyMapping = new LinkedHashMap<>();
+
+        try {
+            Map<String, Object> batchReadApiResp = getBatchDetails(courseId, batchId);
+            if (ObjectUtils.isEmpty(batchReadApiResp)) {
+                logger.info("No batch details found for batchId: {}", batchId);
+                return;
+            }
+            if (StringUtils.isAnyBlank(batchId, profileSurveyId)) {
+                throw new IllegalArgumentException("Batch ID and Profile Survey ID must not be empty.");
+            }
+
+            List<WfStatusEntity> wfStatusEntities = wfStatusEntityRepository.getByApplicationId(batchId);
+            if (CollectionUtils.isEmpty(wfStatusEntities)) {
+                logger.info("No workflow status entities found for batchId: {}", batchId);
+                return;
+            }
+
+            Map<String, Object> surveyResponse = getSurveyResponse(profileSurveyId);
+            Sheet sheet = workbook.createSheet("Enrolment Report");
+
+            Iterator iterator = surveyResponse.entrySet().iterator();
+            Map.Entry<String, Object> firstEntry = (Map.Entry<String, Object>) iterator.next();
+            Map<String, Object> firstResponse = (Map<String, Object>) firstEntry.getValue();
+            Map<String, Object> dataObject = (Map<String, Object>) firstResponse.get(Constants.DATA_OBJECT);
+
+            // Create header row and apply styles
+            createHeaderRow(workbook, sheet, batchReadApiResp, dataObject, headerKeyMapping);
+            int rowNum = 1;
+
+            for (WfStatusEntity wfStatusEntity : wfStatusEntities) {
+                String userId = wfStatusEntity.getUserId();
+                if (StringUtils.isBlank(userId)) {
+                    logger.warn("User ID is blank for WfStatusEntity: {}", wfStatusEntity);
+                    continue;
+                }
+
+                try {
+                    Map<String, Object> propertyMap = new HashMap<>();
+                    propertyMap.put(Constants.ID, userId);
+                    Map<String, Object> userDetails = cassandraOperation.getRecordsByProperties(
+                            Constants.SUNBIRD_KEY_SPACE_NAME, Constants.TABLE_USER, propertyMap, null, Constants.ID);
+                    if (userDetails == null || userDetails.isEmpty()) {
+                        logger.warn("No user details found for userId: {}", userId);
+                        continue;
+                    }
+
+                    Map<String, Object> userInfo = getUserInfo((Map<String, Object>) userDetails.get(userId));
```

Comment:
Instead of reading again from db -- check userUtilityServiceImpl.getUserDetailsFromDB method can be reused here...